### PR TITLE
Fix internal links from Getting Started

### LINKED ISSUE #772: Dead links in new Getting Started
https://bundler.io/guides/getting_started.html, recently generated in #765, has many dead links.

### PR DESCRIPTION
**Pull request by tnir**
- Closes #772
- Fixes up #765

Signed-off-by: Takuya Noguchi [takninnovationresearch@gmail.com](https://github.com/sponsors/tnir)

### CODE REVIEW

**Review by deivid-rodriguez — COMMENTED** [review]
Looks good!

Should we introduce [haml-lint](https://github.com/sds/haml-lint) and split using double quotes and enforcing its usage from now on to a separate PR?

File: source/guides/getting_started.html.haml [hunk 1]
```diff
@@ -13,8 +13,8 @@ title: Getting Started
   Starting work on a project is as simple as <code>bundle install</code>.
 
 .buttons.contents
-  = link_to 'What\'s new in Bundler', './whats_new.html', class: 'btn btn-primary'
-  = link_to 'Why Bundler exists', './rationale.html', class: 'btn btn-primary'
```
Comment (deivid-rodriguez) [inline source/guides/getting_started.html.haml:17]: These are just because it look prettier correct? I would maybe leave them as is if that's the case.

Comment (tnir) [reply source/guides/getting_started.html.haml:17]: Okay, I see.

File: source/guides/getting_started.html.haml [hunk 2]
```diff
@@ -126,15 +126,15 @@ title: Getting Started
...
 .contents
   .buttons
-    = link_to 'Rails', './guides/rails.html', class: 'btn btn-primary'
-    = link_to 'Sinatra', './guides/sinatra.html', class: 'btn btn-primary'
-    = link_to 'RubyGems', './rubygems.html', class: 'btn btn-primary'
-    = link_to 'RubyMotion', './rubymotion.html', class: 'btn btn-primary'
+    = link_to 'Rails', 'rails.html', class: 'btn btn-primary'
+    = link_to 'Sinatra', 'sinatra.html', class: 'btn btn-primary'
+    = link_to 'RubyGems', 'rubygems.html', class: 'btn btn-primary'
+    = link_to 'RubyMotion', 'rubymotion.html', class: 'btn btn-primary'
```
Comment (deivid-rodriguez) [inline source/guides/getting_started.html.haml:137]: Same with these sorry. I personally like the initial dot to emphasize where the relative file lives.

Comment (tnir) [reply source/guides/getting_started.html.haml]: While I do not like the initial dots for the files in the same directory, I do not care about it in this PR.

**Review by deivid-rodriguez — APPROVED** [review]
Awesome, thank you!!

**Pull request merged**